networkx/algorithms/euler.py

```diff
@@ -423,30 +432,51 @@ def eulerize(G):
         raise nx.NetworkXPointlessConcept("Cannot Eulerize null graph")
     if not nx.is_connected(G):
         raise nx.NetworkXError("G is not connected")
-    odd_degree_nodes = [n for n, d in G.degree() if d % 2 == 1]
+    odd_degree_nodes = {n for n, d in G.degree() if d % 2 == 1}
     G = nx.MultiGraph(G)
     if len(odd_degree_nodes) == 0:
         return G
 
-    # get all shortest paths between vertices of odd degree
-    odd_deg_pairs_paths = [
-        (m, {n: nx.shortest_path(G, source=m, target=n)})
-        for m, n in combinations(odd_degree_nodes, 2)
-    ]
-
-    # use inverse path lengths as edge-weights in a new graph
-    # store the paths in the graph for easy indexing later
+    # get all shortest paths lengths between vertices of odd degree
```

Comment:
> I'll reword my question. Have you seen a case when it is faster to use the localized option? Can you find one?

Yes, for example, we can take a large toroidal grid, let's say 500 x 500, `G = nx.grid_2d_graph(500, 500, periodic=True)`
which is Eulerian and add locally some forward diagonals to make it non Eulerian:
```
for i in range(50):
    G.add_edge(2*(i % 10, i // 10), (2 * (i % 10) + 1 , 2 * (i // 10) + 1))
```
The example is a bit extreme, since the odd degree vertices are very close to each other, but it nicely illustrates the difference.
The running time of  `nx.eulerize(G, localized=False)`  was 49.65 sec and for  `nx.eulerize(G, localized=True)`  only 4.76 sec on my machine.

As I mentioned earlier, if there was an option to use a function computing shortest path lengths from a single source to multiple targets then  both localized and non-localized case could be handled more efficiently without this additional parameter.